Debug compilation failure

Compiling with

```
cmake -DSUPPORT_MYSQL=ON -DDEBUG=ON ../
```

fails with the following message

```
vermont/src/common/msg.h:74:109: error: cannot pass objects of non-trivially-copyable type ‘std::string {aka class std::basic_string<char>}’ through ‘...’
#define DPRINTF(fmt, args...) msg2(__LINE__, __FILE__, __PRETTY_FUNCTION__, __func__, MSG_DEBUG, fmt, ##args)
                                                                                                            ^
vermont/src/modules/ipfix/database/IpfixDbWriterSQL.cpp:400:6: note: in expansion of macro ‘DPRINTF’
  DPRINTF("IpfixDbWriter::parseIpfixData: really saw ipfix id %d (%s) in packet with parsedData %llX, type %d, length %d and offset %X", col->ipfixId, ipfix_id_lookup(col->ipfixId, col->enterprise)->name, parsedData, dataTemplateInfo->fieldInfo[k].type.id, dataTemplateInfo->fieldInfo[k].type.length, dataTemplateInfo->fieldInfo[k].offset);
  ^
vermont/src/common/msg.h:74:109: error: cannot pass objects of non-trivially-copyable type ‘std::string {aka class std::basic_string<char>}’ through ‘...’
#define DPRINTF(fmt, args...) msg2(__LINE__, __FILE__, __PRETTY_FUNCTION__, __func__, MSG_DEBUG, fmt, ##args)
                                                                                                            ^
vermont/src/modules/ipfix/database/IpfixDbWriterSQL.cpp:451:3: note: in expansion of macro ‘DPRINTF’
DPRINTF("saw ipfix id %d in packet with parsedData %llX", col->ipfixId, parsedData);
```

--- a/src/modules/ipfix/database/IpfixDbReaderMySQL.cpp
+++ b/src/modules/ipfix/database/IpfixDbReaderMySQL.cpp
@@ -43,11 +43,7 @@ int IpfixDbReaderMySQL::dbReaderSendTable(boost::shared_ptr<TemplateInfo> templa
 	MYSQL_RES* dbResult = NULL;
 	MYSQL_ROW dbRow = NULL;
 	unsigned offset = 0;
-	uint64_t delta = 0;		// 64 bit to avoid castings in the case of flowStartMilliseconds
-	uint32_t flowTime = 0;		// in seconds, so 32 bit are sufficient
-	uint32_t lastFlowTime = 0;
 	uint64_t tmp;
-	bool first = true; 
 	unsigned j = 0;
 	
 	string query = "SELECT " + columnNames + " FROM " + tableName;
@@ -148,7 +144,6 @@ int IpfixDbReaderMySQL::getColumns(const string& tableName)
 	columnNames = "";
 	orderBy = ""; 
 	while((dbRow = mysql_fetch_row(dbResult))) {
-		bool found = true;
 		const struct ipfix_identifier* id = ipfix_name_lookup(dbRow[0]);
 		if (id == NULL) {
 			msg(LOG_NOTICE, "IpfixDbReaderMySQL: Unsupported column: %s", dbRow[0]);

--- a/src/modules/ipfix/database/IpfixDbWriterSQL.cpp
+++ b/src/modules/ipfix/database/IpfixDbWriterSQL.cpp
@@ -303,8 +303,7 @@ void IpfixDbWriterSQL::onDataRecord(IpfixDataRecord* record)
 {
 	// only treat non-Options Data Records (although we cannot be sure that there is a Flow inside)
 	if((record->templateInfo->setId != TemplateInfo::NetflowTemplate)
-		&& (record->templateInfo->setId != TemplateInfo::IpfixTemplate)
-		&& (record->templateInfo->setId != TemplateInfo::IpfixDataTemplate)) {
+		&& (record->templateInfo->setId != TemplateInfo::IpfixTemplate)) {
 		record->removeReference();
 		return;
 	}
@@ -397,17 +396,8 @@ void IpfixDbWriterSQL::fillInsertRow(IpfixRecord::SourceID* sourceID,
 			for(k=0; k < dataTemplateInfo->fieldCount; k++) {
 				if(dataTemplateInfo->fieldInfo[k].type.enterprise == col->enterprise && dataTemplateInfo->fieldInfo[k].type.id == col->ipfixId) {
 					parseIpfixData(dataTemplateInfo->fieldInfo[k].type,(data+dataTemplateInfo->fieldInfo[k].offset), &parsedData);
-					DPRINTF_INFO("IpfixDbWriter::parseIpfixData: really saw ipfix id %d (%s) in packet with parsedData %llX, type %d, length %d and offset %X", col->ipfixId, ipfix_id_lookup(col->ipfixId, col->enterprise)->name, parsedData, dataTemplateInfo->fieldInfo[k].type.id, dataTemplateInfo->fieldInfo[k].type.length, dataTemplateInfo->fieldInfo[k].offset);
+					DPRINTF_INFO("IpfixDbWriter::parseIpfixData: really saw ipfix id %d (%s) in packet with parsedData %p, type %d, length %d and offset %X", col->ipfixId, ipfix_id_lookup(col->ipfixId, col->enterprise)->name, parsedData, dataTemplateInfo->fieldInfo[k].type.id, dataTemplateInfo->fieldInfo[k].type.length, dataTemplateInfo->fieldInfo[k].offset);
 					break;
-				}
-			}
-			// look in static data fields of template for data
-			if (parsedData.empty()) {
-				for(k=0; k < dataTemplateInfo->dataCount; k++) {
-					if(dataTemplateInfo->dataInfo[k].type.enterprise == col->enterprise && dataTemplateInfo->dataInfo[k].type.id == col->ipfixId) {
-						parseIpfixData(dataTemplateInfo->dataInfo[k].type,(dataTemplateInfo->data+dataTemplateInfo->dataInfo[k].offset), &parsedData);
-						break;
-					}
 				}
 			}
 			// check for time-related alternative fields in the database
@@ -448,7 +438,7 @@ void IpfixDbWriterSQL::fillInsertRow(IpfixRecord::SourceID* sourceID,
 				}
 		}
 
-		DPRINTF_INFO("saw ipfix id %d in packet with parsedData %llX", col->ipfixId, parsedData);
+		DPRINTF_INFO("saw ipfix id %d in packet with parsedData %p", col->ipfixId, parsedData);
 
 		if(first) {
 			rowStream << parsedData;

